src/middlewares/User.js

```diff
@@ -30,7 +30,36 @@ const checkEmail = async (req, res, next) => {
   }
 };
 
+const userNameAvailability = async (req, res, next) => {
+  const userConstant = await models.User.findOne({
+    where: {
+      username: req.params.username
+    },
+  });
+    // eslint-disable-next-line no-cond-assign
+  if (userConstant.dataValues.username !== req.params.username) {
```

Comment:
I don't understand the need for this comparison inside this if condition since we have the result of finding one username stored in userConstant variable.

Reply:
explained the reason for using this to Joe.